test/shaping/run-tests.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import sys, os, subprocess
+
+def cmd(command):
+	p = subprocess.Popen (
+		command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+	p.wait ()
+	return p.stdout.read ().decode ('utf-8'), p.returncode
+
+srcdir = os.environ.get ("srcdir", '.')
+builddir = os.environ.get ('builddir', '.')
+top_builddir = os.environ.get ('top_builddir', '../..')
+EXEEXT = os.environ.get ('EXEEXT', '')
+
+extra_options = "--verify"
+hb_shape = top_builddir + "/util/hb-shape" + EXEEXT
+
+fails = 0
+args = sys.argv[1:]
+
+reference = False
+if len (args) and args[0] == "--reference":
+	reference = True
+	args = args[1:]
+
+if not len (args):
+	args = ['/dev/stdin']
+
+for f in args:
+	if not reference:
+		print ("Running tests in " + f)
+
+	for line in open(f).read().strip().splitlines():
```

Comment:
You don’t need to do splitlines to iterate over the lines of a file object, you can just do 

for line in fileobj:

Reply:
Thanks, done.